docs/walkthroughs/developing-apbs-locally.adoc

```diff
@@ -46,6 +46,86 @@ apb init
 
 NOTE: Best practice is to name the APB ending with `-apb`.
 
+== Making your APB consumable as a mobile service
+
+When creating an APB for a mobile service, where possible use Kubernetes resources
+( rather than Openshift resources ) to ensure compatibility with Openshift and Kubernetes.
+
+Only one instance of a service is expected in a namespace. If your service is intended to have more
+than one service instance consideration must be given to the naming of objects.
+
+=== ConfigMaps Vs Secrets
+
+Separate the public and private configuration information in your APB.
+
+Use *configMaps* for public information that a mobile client will need to use the service and label the configMap
+with mobile: enabled and the serviceName so it can be read by the mobile-cli.
+
+An example of this can be seen link:https://github.com/aerogearcatalog/keycloak-apb/blob/master/roles/provision-keycloak-apb/templates/configmap.yml.j2[here].
+
+[source,yaml]
+----
+  labels:
+    mobile: enabled
+    serviceName: my-custom-service
+----
+
+Use *secrets* for private information such as passwords and usernames.
+
+[source,yaml]
+----
+- k8s_v1_secret:
+    name: '{{ myService_secret_name }}'
+    namespace: '{{ namespace }}'
+    labels:
+      name: myService
+      mobile: enabled
+      serviceName: myService
+    string_data:
+      admin_username: '{{ ADMIN_NAME }}'
+      admin_password: '{{ ADMIN_PASSWORD }}'
+----
+
+=== Using the mobile-service tag
+
+For MCP enabled _APBs_, make sure we have a `mobile-service` tag inside the `apb.yml` file:
+
+....
+...
+tags: 
+  - mobile-service
+....
+
+=== Using the servicename label
+
+Inside of the `apb.yml` file, make sure you use the `serviceName:` label, like:
+
+....
+...
+metadata:
+  displayName: Aerogear Sync Server
+  console.openshift.io/iconClass: icon-nodejs
+  serviceName: fh-sync-server
+...
+....
+
+=== Integrating with the metrics service
+
+To allow Prometheus to auto discover your custom services' metrics endpoint you need to include an annotation when creating the 
+Kubernetes service in your provisioning task.
+
+[source,yaml]
+----
+annotations:
+  org.aerogear.metrics/plain_endpoint: /my-metrics-endpoint
```

Comment:
Where does `/my-metrics-endpoint` come from?

Reply:
I just used it as a stand in. Right after this I linked to the keycloak metrics end-point in the keycloak config. I just didn't want to specify something in case a developer reading it then thought that they have to use the same route when that's not the case.